src/js/profile/mobile/widget/Calendar.js

```diff
@@ -0,0 +1,835 @@
+									this.contDates.childNodes[oneWeek].childNodes[dayCount].cssDate = (!this._nullDate && allDaysToTime == timeToNumber);
+									this.contDates.childNodes[oneWeek].childNodes[dayCount].cssWeekend = (weekStart >= 7);
+									this._updateCellStyle(oneWeek, dayCount, false);
+									if (allDaysToTime == timeToNumber) {
+										this._activeDateCell = this.contDates.childNodes[oneWeek].childNodes[dayCount];
+										this._updateCellStyle(oneWeek, dayCount, true);
+									}
+									if (++weekStart > 7) {
+										weekStart = 1;
+									}
+									addMonthCount++;
+								}
+							}
+						}
+						this.contMonth.firstChild.firstChild.childNodes[1].innerHTML = this._buildMonthHdr(dateFullFormat);
+					};
+					this._updateCellStyle = function (col, row, flag) {
+						var selectDate = this.contDates.childNodes[col].childNodes[row];
+						var cellStr = "calendar_cell calendar_cell";
+
+						cellStr += (selectDate.cssMonth ? "_month" : "");
+						cellStr += (selectDate.cssDate ? "_date" : "");
+						cellStr += (selectDate.cssWeekend ? "_weekend" : "");
+						selectDate.className = cellStr;
+						var child = selectDate.getElementsByTagName("div")[0];
+
+						if (flag === false) {
+							child.style.backgroundColor = "#FFFFFF";
+						} else {
+							child.style.backgroundColor = "#0381fe";
+							child.style.width = "28px";
+							child.style.height = "28px";
+							child.style.margin = "auto";
+							child.style.borderRadius = "50%";
+						}
+						selectDate = null;
+					};
+
+					this._isSelectorVisible = function () {
+						if (!this._sel) {
+							return false;
+						}
+						return (this._sel.style.display != "none");
+					};
+					this._doOnSelectorChange = function () {
+						this.callEvent("onChange",
+							[new Date(this._activeMonth.getFullYear(),
+							this._activeMonth.getMonth(),
+							this._activeDate.getDate(),
+							this._activeDate.getHours(),
+							this._activeDate.getMinutes(),
+							this._activeDate.getSeconds())])
+					};
+
+					this._dateFormat = "";
+					this._dateFormatRE = null;
+					this.setDateFormat = function (dateFormat) {
+						var obj = {};
+
+						if (this._strToDate != null) {
+							for (var idx in this.globalObj) {
+								if (this.globalObj[idx].input != null && this.globalObj[idx].input.value.length > 0) {
+									var convertDate = this._strToDate(this.globalObj[idx].input.value, this._dateFormat || this.langData[this.lang].dateformat || "%Y-%m-%d");
+
+									if (convertDate instanceof Date) {
+										obj[idx] = convertDate;
+									}
+								}
+							}
+						}
+						this._dateFormat = dateFormat;
+						var regexpFunc = String(this._dateFormat).replace(/%[a-zA-Z]+/g, function (str) {
+							var inputCase = str.replace(/%/, "");
+
+							switch (inputCase) {
+								case "n":
+								case "h":
+								case "j":
+								case "g":
+								case "G":
+									return "\\d{1,2}";
```

Comment:
What's means the "n", "h","j","g","G" and why do you return the value??

Reply:
h, g, G is not necessary input case.
I will remove them.

and the other case is here,

case "%d":
										return timeChecker(timeNdate.getDate());
									case "%j":
										return timeNdate.getDate();
									case "%D":
										return dateType.dayShortName[timeNdate.getDay()];
									case "%l":
										return dateType.dayFullName[timeNdate.getDay()];
									case "%m":
										return timeChecker(timeNdate.getMonth() + 1);
									case "%n":
										return timeNdate.getMonth() + 1;
									case "%M":
										return dateType.monthShortName[timeNdate.getMonth()];
									case "%F":
										return dateType.monthFullName[timeNdate.getMonth()];
									case "%y":
										return timeChecker(timeNdate.getYear() % 100);
									case "%Y":
										return timeNdate.getFullYear();
									case "%%":
										return "%";
									case "%P":
...